src/components/Store.react.js

```diff
@@ -186,11 +153,19 @@ export default class Store extends React.Component {
         }
         const old = this._backstore.getItem(id);
         // Only set the data if it's not the same data.
-        if (dataChanged(data, old)) {
-            this._backstore.setItem(id, data);
-            setProps({
-                modified_timestamp: this._backstore.getModified(id),
-            });
+        // If the new data is undefined, we got here by overwriting the entire
+        // component with a new copy that has no `data` specified - so pull back
+        // out the old value.
+        // Note: this still allows you to set data to null
+        if (!equals(data, old)) {
+            if (data === undefined) {
+                setProps({data: old});
```

Comment:
Only setting `data` here prevents scenarios using `Input=modified_timestamp, State=data --> Output=data` from being triggered. Could we add `modified_timestamp: this._backstore.getModified(id),` here too?